wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/shuffleboard/Robot.java

```diff
@@ -0,0 +1,65 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.examples.shuffleboard;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableType;
+import edu.wpi.first.wpilibj.AnalogPotentiometer;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.IterativeRobot;
+import edu.wpi.first.wpilibj.Spark;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+
+//import java.util.Map;
+
+@SuppressWarnings({"MemberName", "LineLength", "RegexpSinglelinejava"})
+public class Robot extends IterativeRobot {
+  private final DifferentialDrive tankDrive = new DifferentialDrive(new Spark(0), new Spark(1));
+  private final Encoder leftEncoder = new Encoder(0, 1);
+  private final Encoder rightEncoder = new Encoder(2, 3);
+
+  private final Spark elevatorMotor = new Spark(2);
+  private final AnalogPotentiometer elevatorPot = new AnalogPotentiometer(0);
+  private NetworkTableEntry maxSpeed;
+
+  @Override
+  public void robotInit() {
+    // Add a 'max speed' widget to a tab named 'Configuration', using a number slider
+    maxSpeed = Shuffleboard.add("Max Speed", NetworkTableType.kDouble)
+        .toTab("Configuration")
+        .withWidget("Number Slider")
+        //.withProperties(Map.of("min", 0, "max", 1)) // Map.of introduced in Java 9
+        .getEntry();
+
+    // Add the tank drive and encoders to a 'Drivebase' tab
+    Shuffleboard.add("Tank Drive", tankDrive)
+        .toTab("Drivebase");
+    // Put both encoders in a list layout
+    Shuffleboard.add("Left Encoder", leftEncoder)
+        .toTab("Drivebase")
+        .toLayout("List", "Encoders");
+    Shuffleboard.add("Right Encoder", rightEncoder)
+        .toTab("Drivebase")
+        .toLayout("List", "Encoders");
```

Comment:
Not sure if I like the way this reads because of the duplication.  Is it also another different paradigm for accessing multiple levels of network table compared to ntcore.  Can we change the way this works to be closer to the way people work with ntcore?
```
ShuffleboardTab tab = Shuffleboard.getTab("Drivebase");
ShuffleboardLayout encoderList = tab.getLayout("List", "Encoders") // It'd be nice if List was not a string and rather an enum value too.
encoderList.add("Left Encoder", leftEncoder);
encoderList.add("Right Encoder", rightEncoder);
```
or
```
Shuffleboard.getTab("Drivebase")
  .getLayout("List", "Encoders")
  .add("Left Encoder", leftEncoder)
  .add("Right Encoder", rightEncoder);
```

Reply:
I think that would make sense. The second one could be shorthand for the first, avoiding the need to keep references around to intermediate containers if it's not needed.

Layouts would not be possible to define in an enum, since they are not constant and may be available in third-party plugins, so we would have to mix-and-match enums and string constants for base plugin layouts and third-party layouts, respectively. I prefer the consistency of only using string constants